### PR TITLE
fix: various auto-populate improvements

### PR DESCRIPTION
**Pull request by axelstudios**
#### Any background context you want to provide?
The auto-populate functionality has already been merged, but this PR adds a number of small improvements

#### What's this PR do?
- Backend verification that the cycle doesn't contain any properties or taxlots
- Fixed merge logic and removed duplicate from merge
- Frontend error notification if no cycles exist
- Call `insert_sample_data` before fetching columns since that process creates one `Note` extra_data column.  This is for the Show Only Populated Columns functionality
- Moves data to insert into separate json file
- Creates a TaxlotAuditLog record for the one taxlot record

#### How should this be manually tested?
- Delete all cycles and attempt to open the modal
- Test the `insert_sample_data` API call for an org that already contains data

#### What are the relevant tickets?
#2595

### CODE REVIEW

File: seed/views/v3/organizations.py [hunk 1]
```diff
@@ -1702,7 +1483,8 @@ def insert_sample_data(self, request, pk=None):
             geocode = PropertyState.objects.filter(id__in=ids)
             geocode_buildings(geocode)
 
-        # Create a merge of 2 properties
+        # Create a merge of the last 2 properties
+        state_ids_to_merge = ids[-2:]
```
Comment (macintoshpie) [inline seed/views/v3/organizations.py:1487]: 🆒

File: seed/views/v3/organizations.py [hunk 2]
```diff
@@ -1722,12 +1504,12 @@ def insert_sample_data(self, request, pk=None):
         import_record = ImportRecord.objects.create(name='Auto-Populate', super_organization=org)
 
         # Interval Data
-        filename = "example-pm-monthly-meter-usage.xlsx"
-        filepath = os.path.dirname(os.path.abspath(__file__)) + "/data/" + filename
+        filename = 'example-pm-monthly-meter-usage.xlsx'
+        filepath = f"{Path(__file__).parent.absolute()}/data/{filename}"
```
Comment (macintoshpie) [inline seed/views/v3/organizations.py:1508]: you probably know this already, but I learned you could also use `/` and strings to create paths (no better, just thought I'd mention it)
```python
str((Path(__file__).parent / 'data' / filename).absolute())
```

**Review by macintoshpie — APPROVED** [review]
Looks good to me! Tested both with cycle with data and no cycles and got the expected errors.
Screen Shot 2021-04-08 at 9 12 05 AM

**Review by adrian-lara — APPROVED** [review]
I only tested the items I left comments about (tax lot audit and no cycle exists bug), but they work well!

Code changes related to those two look good, and the rest of the code changes look reasonable! 👍🏽

File: seed/views/v3/organizations.py (hunk 2 above)
Comment (axelstudios) [reply seed/views/v3/organizations.py:1508]: I haven't seen that, but that's a cool operator override

**Pull request merged**